### PR TITLE
Add paging to plagiarism results

### PR DESCRIPTION
**Pull request by philippbauch**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Client: I followed the [coding and design guidelines](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/client.html).
- ~[ ] Client: I added multiple integration tests (Jest) related to the features (with a high test coverage)~
- [x] Client: I documented the TypeScript code using JSDoc style.
- [x] Client: I added multiple screenshots/screencasts of my UI changes

### Motivation and Context
For exercises with a large amount of submissions, displaying all detected plagiarism matches is quite expensive and slows down the client application. This is why this PR adds paging functionality to show a max. number of 100 matches at once.

### Description
I added a small paging panel at the bottom of the sidebar to navigate through multiple pages. Each page will show 100 results.

### Steps for Testing

**Note**: For testing purposes, I decreased the paging size to 10.

The following should work for **modeling**, **text**, and **programming** exercises. Make sure you have multiple similar submissions to generate enough results.

1. Log in to Artemis
2. Navigate to Course Administration
3. Select a course
4. Go to Exercises
5. Navigate to the details page of an exercise by clicking 'View'
6. Click 'Check Plagiarism'
7. Click 'Check Plagiarism' on the plagiarism page and wait for the results
8. If you have more than 10 results, they should be paged correctly and only 10 results should be displayed at once

### Test Coverage
* TODO: plagiarism-sidebar.component.spec.ts 25%

### Screenshots
Screenshot 2020-12-22 at 22 36 46

### CODE REVIEW

**Review by airelawaleria — APPROVED** [review]
Tested on ts1 with @simonzilker @JohannesWeiss @EneaGore @marvinchimeka .
Works fine!

**Review by JohannesWeiss — DISMISSED** [review]
Tested in testing session and works as intended with the pages.

**Review by simonzilker — DISMISSED** [review]
Works perfectly on ts1!

**Review by marvinchimeka — APPROVED** [review]
tested in testing session. Approve

**Review by EneaGore — APPROVED** [review]
Testing session on ts1, workds as expected.

**Review by krusche — DISMISSED** [review]

File: src/main/webapp/app/exercises/shared/plagiarism/plagiarism-sidebar/plagiarism-sidebar.component.ts [hunk 1]
```diff
@@ -8,9 +8,71 @@ import { ModelingSubmissionElement } from 'app/exercises/shared/plagiarism/types
...
+
+    /**
+     * Subset of currently paged comparisons.
+     */
+    public pagedComparisons?: PlagiarismComparison<TextSubmissionElement | ModelingSubmissionElement>[];
+
+    /**
+     * Number of comparisons per page.
+     */
+    public pageSize = 10;
```
Comment (krusche) [inline src/main/webapp/app/exercises/shared/plagiarism/plagiarism-sidebar/plagiarism-sidebar.component.ts:35]:
```suggestion
    public pageSize = 100;
```

**Pull request merged**